### ISSUE TITLE
Cant pip install mediapipe

### ISSUE THREAD

**Issue by nikitatishin5** [google/mediapipe#1325]
I cant install mediapipe in Python
There is no such a package

**Comment by jiuqiant** [comment]
https://pypi.org/project/mediapipe/

**Comment by jiuqiant** [comment]
You can find it at https://pypi.org/project/mediapipe.

**Comment by MaRauder111** [comment]
I having some error in "Ubuntu 18.04.1"

Creating a virtual environment name "env"

``` 
python3 -m venv env

source env/bin/activate
```
Installing mediapipe
```
(env) lamzing@lamzing-Lenovo-G50-70:~/Mediapipe_testing/Testing/8/gesture_recognition-main$ pip install mediapipe
Collecting mediapipe
  Could not find a version that satisfies the requirement mediapipe (from versions: )
No matching distribution found for mediapipe

```

**Comment by jiuqiant** [comment]
@MaRauder111, what's your python version?

**Comment by MaRauder111** [comment]
My python version was 2. I have upgrade my python.

***Before***
```
lamzing@lamzing-Lenovo-G50-70:~$ python --version
Python 2.7.0
```

***Now***
```
lamzing@lamzing-Lenovo-G50-70:~$ python --version
Python 3.7.9
```

Thank you @jiuqiant . I totally forget to check about the python version. My bad.

**Comment by jiuqiant** [comment]
You are welcome!

**Comment by nurettin** [comment]
Also note that pip install will currently not work with versions >= 3.9

**Comment by Makuta11** [comment]
I am getting the same error as MaRauder111 on Ubuntu 20.04:

(env) ubuntu@ubuntu:~/mediapipe$ python --version
Python 3.8.5
(env) ubuntu@ubuntu:~/mediapipe$ pip install mediapipe
ERROR: Could not find a version that satisfies the requirement mediapipe (from versions: none)
ERROR: No matching distribution found for mediapipe

Anyone know a fix?

**Comment by MaRauder111** [comment]
> I am getting the same error as MaRauder111 on Ubuntu 20.04:
> 
> (env) ubuntu@ubuntu:~/mediapipe$ python --version Python 3.8.5 (env) ubuntu@ubuntu:~/mediapipe$ pip install mediapipe
> ERROR: Could not find a version that satisfies the requirement mediapipe (from versions: none)
> ERROR: No matching distribution found for mediapipe
> 
> Anyone know a fix?

I update my python to 3 and it work fine for me.

**Comment by Makuta11** [comment]
> > I am getting the same error as MaRauder111 on Ubuntu 20.04:
> > (env) ubuntu@ubuntu:~/mediapipe$ python --version Python 3.8.5 (env) ubuntu@ubuntu:~/mediapipe$ pip install mediapipe
> > ERROR: Could not find a version that satisfies the requirement mediapipe (from versions: none)
> > ERROR: No matching distribution found for mediapipe
> > Anyone know a fix?
> 
> I update my python to 3 and it work fine for me.

I tried that and got the following:

ubuntu@ubuntu:~$ python3 -m venv env
ubuntu@ubuntu:~$ source env/bin/activate
(env) ubuntu@ubuntu:~$ python --version
Python 3.8.5
(env) ubuntu@ubuntu:~$ pip install mediapipe
ERROR: Could not find a version that satisfies the requirement mediapipe (from versions: none)
ERROR: No matching distribution found for mediapipe

any ideas?

**Comment by mayank6940** [comment]
**I have python 3.8.3 version but I can't install mediapipe it's shows me Could not find a version that satisfies the requirement mediapipe

**Comment by gamersi** [comment]
@mayank6940 I gat this, too, i have python 3.8.5

**Comment by mayank6940** [comment]
@gamersi  install python 3.7 version then you will not get any error

**Comment by gamersi** [comment]
> @gamersi  install python 3.7 version then you will not get any error 

Ok I will try it later

**Comment by gamersi** [comment]
@mayank6940 I am now using the 3.7.9 version with PIP 21.0.1, I deleted 3.8 and i still get the same error

**Comment by gamersi** [comment]
Like this:
![image](https://user-images.githubusercontent.com/46320904/114303419-52525500-9ace-11eb-98f3-11212c5e5ccf.png)

**Comment by shabelson** [comment]
Hi - Same issue.
tried on python from 3.2 to 3.9 (The image is on 3.7)
Same as with anaconda...
![Capture](https://user-images.githubusercontent.com/23707195/115282799-e6fe2800-a14a-11eb-975b-649da3bb662f.JPG)

**Comment by gamersi** [comment]
> Hi - Same issue.
> 
> tried on python from 3.2 to 3.9 (The image is on 3.7)
> 
> Same as with anaconda...
> 
> ![Capture](https://user-images.githubusercontent.com/23707195/115282799-e6fe2800-a14a-11eb-975b-649da3bb662f.JPG)
> 
> 

I still didn't find a solution - python makes problems everytime

**Comment by jiuqiant** [comment]
> > Hi - Same issue.
> > tried on python from 3.2 to 3.9 (The image is on 3.7)
> > Same as with anaconda...
> > ![Capture](https://user-images.githubusercontent.com/23707195/115282799-e6fe2800-a14a-11eb-975b-649da3bb662f.JPG)
> 
> I still didn't find a solution - python makes problems everytime

I think "python37-32" means **32-bit** ? Your pip is for 32-bit version of Python? 
You need a 64-bit version of Python and Pip to install mediapipe...

**Comment by IsakKruger** [comment]
If you get:
ERROR: Could not find a version that satisfies the requirement mediapipe
ERROR: No matching distribution found for mediapipe
when trying "pip install mediapie"

Try "py -m pip install mediapie" insted, worked for me!

**Comment by Rhytham3008** [comment]
it didn't worked for me it says py is not recognised as an internal or external command

**Comment by Rhytham3008** [comment]
pls help

**Comment by jiuqiant** [comment]
> it didn't worked for me it says py is not recognised as an internal or external command

py is probably an alias to the local python 3.7+ binary.
Also see the sys requirements at https://github.com/google/mediapipe/blob/master/docs/getting_started/troubleshooting.md#python-pip-install-failure.

**Comment by JainJai2305** [comment]
Hey guys !! those of u who are not able to install mediapipe and it's showing error like this  "No matching distribution found for mediapipe" ;  install python 3.7.0 not 3.7.4 not 3.8  ,  Python 3.7.0  link:   https://www.python.org/downloads/release/python-370/

I had 3.7.4 but I was not able to install mediapipe , but I changed my path and installed 3.7.0 and now it's working just fine for me 
Hope it helped , luck!! :)

**Comment by Nostalgie2** [comment]
I am trying pip install mediapipe but I am getting an error. ERROR: could not find a version that satisfies the requierement mediapipe. no matching distribution found for mediapipe.

**Comment by jiuqiant** [comment]
> I am trying pip install mediapipe but I am getting an error. ERROR: could not find a version that satisfies the requierement mediapipe. no matching distribution found for mediapipe.

Does https://github.com/google/mediapipe/blob/master/docs/getting_started/troubleshooting.md#python-pip-install-failure help?

**Comment by pandeyshambhavi** [comment]
I have python version 3.7.6 still it shows ERROR: could not find a version that satisfies the requierement mediapipe. no matching distribution found for mediapipe.

**Comment by jiuqiant** [comment]
> I have python version 3.7.6 still it shows ERROR: could not find a version that satisfies the requierement mediapipe. no matching distribution found for mediapipe.

What's your OS and have you checked your python 3.7.6 and pip are the 64-bit version?

**Comment by pandeyshambhavi** [comment]
Yes I downloaded the 64 bit version of python 3.7.6 and finally medipipe is installed. Thank you!

**Comment by Rhytham3008** [comment]
> > I am trying pip install mediapipe but I am getting an error. ERROR: could not find a version that satisfies the requirement mediapipe. no matching distribution found for mediapipe.
> 
> Does https://github.com/google/mediapipe/blob/master/docs/getting_started/troubleshooting.md#python-pip-install-failure help?

Thanks, I got to know that it needs python to be 64 bit

**Comment by Missael512** [comment]
Hello guys, i'm having the same problem as others: 
ERROR: Could not find a version that satisfies the requirement mediapie (from versions: none)
ERROR: No matching distribution found for mediapie
I have Python 3.8.5 running on a 64 bits System , i don't know what to do

**Comment by chiazor** [comment]
> Hey guys !! those of u who are not able to install mediapipe and it's showing error like this "No matching distribution found for mediapipe" ; install python 3.7.0 not 3.7.4 not 3.8 , Python 3.7.0 link: https://www.python.org/downloads/release/python-370/
> 
> I had 3.7.4 but I was not able to install mediapipe , but I changed my path and installed 3.7.0 and now it's working just fine for me
> Hope it helped , luck!! :)

This exactly worked for me too. I downloaded python 3.7 and created a virtualenv with 3.7. 
**virtualenv venv --python=python3.7.0**

**Comment by mayank6940** [comment]
> Hello guys, i'm having the same problem as others:
> ERROR: Could not find a version that satisfies the requirement mediapie (from versions: none)
> ERROR: No matching distribution found for mediapie
> I have Python 3.8.5 running on a 64 bits System , i don't know what to do

install python 3.7 because mediapipe only works on python 3.7...

**Comment by iovsiann** [comment]
Below worked successfully for me on Ubuntu 18.04:
````
sudo apt update 
sudo apt install software-properties-common
sudo add-apt-repository ppa:deadsnakes/ppa 
sudo apt update 
sudo apt install python3.9 python3.9-venv
python3.9 -m venv mp && source mp/bin/activate
pip install mediapipe
Collecting mediapipe
  Downloading mediapipe-0.8.6.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (32.2 MB)
````

**Comment by adamanov** [comment]
for **Ubuntu 18**
you have to just upgrade your pip with the following command

`python -m pip install --upgrade pip` 
`pip install mediapipe`

**Comment by gh279760559** [comment]
> for **Ubuntu 18**
> you have to just upgrade your pip with the following command
> 
> `python -m pip install --upgrade pip`
> `pip install mediapipe`

Guys, python version does not matter too much. Check @adamanov comments. If you upgrade your pip version it should be fine. Thanks Adam

**Comment by agneya-1402** [comment]
ERROR: Could not find a version that satisfies the requirement mediapipe (from versions: none)
ERROR: No matching distribution found for mediapipe

Using Python 3.10 on windows 10
Command used : python -m pip install mediapipe

Anyone got a fix ?

**Comment by shabelson** [comment]
the best practice I know is to :
1. Download python 3.7 (it can be in parallel to other python versions on the same machine)
2. create a virtual environment (several options - tutorial easy and available)
3. activate the venv and pip install on the said environment.
Good Luck!

**Comment by FAHMYPAIZIAH** [comment]
so I can't download mediapipe in python 3.10

**Comment by FAHMYPAIZIAH** [comment]
> > Help :(
> > ![image](https://user-images.githubusercontent.com/90369043/138430995-fe588ed0-18f6-4fc6-9dbd-5c8a48828972.png)
> > **Windows 11**
> 
> Python 3.10 is not supported by Mediapipe yet. Please use Python 3.7 to 3.9.

Same problem :)

**Comment by shivtiwari123** [comment]
> > > Help :(
> > > ![image](https://user-images.githubusercontent.com/90369043/138430995-fe588ed0-18f6-4fc6-9dbd-5c8a48828972.png)
> > > **Windows 11**
> > 
> > 
> > Python 3.10 is not supported by Mediapipe yet. Please use Python 3.7 to 3.9.
> 
> Same problem :)

Tried python 3.7.9. Was able to install mediapipe., but numpy and cv2 went invalid and could not be used my previous codes.

**Comment by rodrigoricky** [comment]
I tried downgrading from v3.10 to v3.7 and it successfully installed.

**Comment by edulayernet** [comment]
Guys, just type enter this "pip install mediapipe==0.8.9" to bash or cmd.

**Comment by rodrigoricky** [comment]
Updated to python 3.8, can't download it even @edulayernet  solution, no worky.

**Comment by Rana-Banerjee** [comment]
Tried python 3.7.5 and 3.8.0, but pip install mediapipe doesn't work. Says no matching distributions found. Building from source also failed with the following message: 
mediapipe/calculators/tensor/image_to_tensor_converter_opencv.cc: In member function 'virtual absl::lts_20210324::StatusOr<mediapipe::Tensor> mediapipe::{anonymous}::OpenCvProcessor::Convert(const mediapipe::Image&, const mediapipe::RotatedRect&, const mediapipe::Size&, float, float)':
mediapipe/calculators/tensor/image_to_tensor_converter_opencv.cc:106:12: error: could not convert 'tensor' from 'mediapipe::Tensor' to 'absl::lts_20210324::StatusOr<mediapipe::Tensor>'
     return tensor;
            ^~~~~~
Target //mediapipe/modules/face_detection:face_detection_short_range_cpu failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 22.216s, Critical Path: 21.19s
INFO: 187 processes: 129 internal, 58 linux-sandbox.
FAILED: Build did NOT complete successfully

**Comment by rafiizdhr** [comment]
<img width="668" alt="image" src="https://user-images.githubusercontent.com/93192755/144746156-a83ca3a8-7443-447f-b68e-7b29417c5976.png">
i use python 3.10.0
i cant install mediapipe, is there any answer for this?

**Comment by rodrigoricky** [comment]
I just realized I was using python 32bit, after I switched from Python 32bit to 63bit, Mediapipe should work from Python 7 to 9 64bit.

**Comment by ALISONTORRES** [comment]
C:\Users\Alison>python --version
Python 3.10.0

C:\Users\Alison>pip install mediapipe
ERROR: Could not find a version that satisfies the requirement mediapipe (from versions: none)
ERROR: No matching distribution found for mediapipe

C:\Users\Alison>
 same problem :( help

**Comment by rodrigoricky** [comment]
> C:\Users\Alison>python --version Python 3.10.0
> 
> C:\Users\Alison>pip install mediapipe ERROR: Could not find a version that satisfies the requirement mediapipe (from versions: none) ERROR: No matching distribution found for mediapipe
> 
> C:\Users\Alison> same problem :( help

Use either Python 3.7 to 3.9, and make sure it's 64bit

**Comment by sgowroji** [comment]
Python 3.10 is now officially supported #2884 (comment)

**Comment by CZ0520** [comment]
it still doesn't work for me on python 3.10:
<img width="570" alt="Screen Shot 2021-12-20 at 3 31 15 PM" src="https://user-images.githubusercontent.com/94819198/146731635-c23932d7-2b9c-4b81-a231-b544513e8d16.png">

**Comment by wu-yakun** [comment]
/home/nvidia/Pictures/Screenshot from 2021-12-28 18-46-53.png

**Comment by wu-yakun** [comment]
If your processor is arm,if can not be downloaded, mediapipe only supports x86 and amd architecture!
![Screenshot from 2021-12-28 18-46-53](https://user-images.githubusercontent.com/48200202/147559117-79c12be2-fbfb-4d50-abc7-219698f288ed.png)

**Comment by CZ0520** [comment]
thanks that helped a lot!

**Comment by alanb43** [comment]
Darn, any possibility they add compatibility for ARM based systems? All M1 macbooks are out of the question :(

**Comment by Hitheshgit** [comment]
> py -m pip install mediapie

Hey did not work for me.Please help

**Comment by shanggangli** [comment]
> 

can't  work.

**Comment by leo-smi** [comment]
guys, try this:

```pip3 install mediapipe --user```

**Comment by root-master** [comment]
There is something wrong with the build process. `pip install mediapipe` works neither on intel mac nor on M1. I tried all the suggestions.

**Comment by Bruno-Jang** [comment]
> If you get: ERROR: Could not find a version that satisfies the requirement mediapipe ERROR: No matching distribution found for mediapipe when trying "pip install mediapie"
> 
> Try "py -m pip install mediapie" insted, worked for me on windows!

Thanks!
It worked!
For me "python3.8 -m pip install mediapipe==0.8.4"
You saved my time..!!

**Comment by yang242j** [comment]
> suggestions

Try install mediapipe-silicon, for m1 Mac user

**Comment by 22pilarskil** [comment]
> > suggestions
> 
> Try install mediapipe-silicon, for m1 Mac user
> 
> FYI, mediapipe only support 64bit for now, not 32bit [Youtube video proof](https://www.youtube.com/watch?v=4yyey-dCG1I)

This worked for me with an M1 MacAir, tysm

**Comment by younasm** [comment]
> 

did you find any solution?

**Comment by Psychemist** [comment]
> 

Thanks!

`pip install mediapipe-silicon` 
This cmd works on my M1 MacBook

**Comment by younasm** [comment]
> > 
> 
> Thanks!
> 
> `pip install mediapipe-silicon` This cmd works on my M1 MacBook

yes it's installed but not perfect for my case.

**Comment by zan-dong** [comment]
> Hey guys !! those of u who are not able to install mediapipe and it's showing error like this "No matching distribution found for mediapipe" ; install python 3.7.0 not 3.7.4 not 3.8 , Python 3.7.0 link: https://www.python.org/downloads/release/python-370/
> 
> I had 3.7.4 but I was not able to install mediapipe , but I changed my path and installed 3.7.0 and now it's working just fine for me Hope it helped , luck!! :)

This really help me sooo much!thx

**Comment by alabamustapha** [comment]
1. Make sure you are using python 3.7; you can download it [here](https://www.python.org/downloads/release/python-370/)  
2. If you would still like to maintain multiple installations of python, follow this [article](https://levelup.gitconnected.com/how-to-install-and-manage-multiple-python-versions-on-windows-10-c90098d7ba5a)

**Comment by BasittyAnjali** [comment]
pip install mediapipe
Defaulting to user installation because normal site-packages is not writeable
ERROR: Could not find a version that satisfies the requirement mediapipe (from versions: none)
ERROR: No matching distribution found for mediapipe

same error .
plz help

**Comment by agiamsoftech** [comment]
No any one fixed this issues?
[
![image](https://user-images.githubusercontent.com/117843343/207899897-d81a0a8a-e36e-4737-a509-80821573f89b.png)
![Screenshot_2](https://user-images.githubusercontent.com/117843343/207899969-00b86cef-42df-4b2d-a81d-0e522408caf7.png)
](url)

**Comment by agiamsoftech** [comment]
No any one fixed issues?
I'm used all type of command in my terminal but problem not fix.
![Uploading Screenshot_3.png…]()

**Comment by agiamsoftech** [comment]
No any one fixed issues?
I'm used all type of command in my terminal but problem not fix.
![Screenshot_3](https://user-images.githubusercontent.com/117843343/207901119-f1b799d6-c46a-4129-8b39-038f186d9f28.png)

**Comment by RobotRoss** [comment]
Still an issue

**Comment by jabedkhanjb** [comment]
i've got the same issues, can't fix it. please assist me
![image](https://user-images.githubusercontent.com/86038917/215288289-93bc4f09-4468-4bb4-a048-dc8727dbd4cd.png)

**Comment by Caraffa-git** [comment]
Still
![image](https://user-images.githubusercontent.com/70948486/216620528-d60aefcb-18c5-4077-baf7-534bb3241eb3.png)

**Comment by Sanimag** [comment]
Hello, same issue
![image](https://user-images.githubusercontent.com/56559719/217215931-262025bc-35ab-43f4-b2c9-61f2913a10d8.png)
![image](https://user-images.githubusercontent.com/56559719/217216015-66d5dc7b-8fc6-4d54-8594-eed285e31f8b.png)
![image](https://user-images.githubusercontent.com/56559719/217216079-7a27db8b-20fb-40c5-92e0-891970ab68e1.png)
![image](https://user-images.githubusercontent.com/56559719/217216181-01aab10e-28c6-44e0-9b18-d052adaa7dda.png)
![image](https://user-images.githubusercontent.com/56559719/217216236-e1755dfb-a459-4fa3-ac27-f7ae62af2f6c.png)

**Comment by nurnahian** [comment]
![image](https://user-images.githubusercontent.com/52484826/219714927-07b19a74-0caf-46a1-87a5-cd8a00f1643b.png)
Python 3.11.2
pip 23.0

**Comment by Mitalikodkani** [comment]
> ![image](https://user-images.githubusercontent.com/52484826/219714927-07b19a74-0caf-46a1-87a5-cd8a00f1643b.png) Python 3.11.2 pip 23.0

only python versions 3.7-3.10 support mediapipe. 3.11 doesn't. So download a 3.10 version of python, that should work immediately

**Comment by Recolet-a** [comment]
> > ![image](https://user-images.githubusercontent.com/52484826/219714927-07b19a74-0caf-46a1-87a5-cd8a00f1643b.png) Python 3.11.2 pip 23.0
> 
> only python versions 3.7-3.10 support mediapipe. 3.11 doesn't. So download a 3.10 version of python, that should work immediately

I read the whole string of comments to finally discover that I was supposed to use 3.10.

**Comment by NoureddinSameer** [comment]
try this command 
pip install mediapipe-rpi4

**Comment by mdjavedakhtar** [comment]
> > > ![image](https://user-images.githubusercontent.com/52484826/219714927-07b19a74-0caf-46a1-87a5-cd8a00f1643b.png) Python 3.11.2 pip 23.0
> > 
> > 
> > only python versions 3.7-3.10 support mediapipe. 3.11 doesn't. So download a 3.10 version of python, that should work immediately
> 
> I read the whole string of comments to finally discover that I was supposed to use 3.10.

Works perfectly !!

**Comment by Ayad-Mihidabi-Khan-Jitu** [comment]
![image](https://user-images.githubusercontent.com/66487318/226744132-968abdd7-90f2-4a08-97a5-b1c761129e54.png)
Still does not work

**Comment by deusro** [comment]
it worked with version 3.10.7

**Comment by mdjavedakhtar** [comment]
With v 3.10.0 i was able to get it to work

**Comment by osssama98** [comment]
this is weird I just make another environment with both 3.10.10 and 3.10.0 and still doesn't work.

> With v 3.10.0 i was able to get it to work



> it worked with version 3.10.7

**Comment by SivaRamana-H-V** [comment]
Thank You @Suchir2004  It's Work Properly but it supported `pip install mediapipe` this Module

**Comment by Suchir2004** [comment]
See @SivaRamana-H-V  
When I am installing this module in old pc it give the same error but when I install this module using pycharm in my friend pc it installed easily and doesn't give any error. So I think this module require a updated version of python to install.

**Comment by Suchir2004** [comment]
@mahesh021011 Yes bro I tried it too but not working...

**Comment by Ayad-Mihidabi-Khan-Jitu** [comment]
> 
this Python Package (Unofficial) is for RaspberryPi.
will this work for windows!!??

**Comment by Ayad-Mihidabi-Khan-Jitu** [comment]
> Use pip install ffmpeg python3 opencv
> 
> pip install mediapipe-rpi4

this Python Package (Unofficial) is for RaspberryPi.
will this work for windows!!??

**Comment by Suchir2004** [comment]
Download the mediapipe file manually and then copy the  path to your cmd terminal. Then you'll find that you installed mediapipe. Works on windows 64 bits

**Comment by Ayad-Mihidabi-Khan-Jitu** [comment]
> Download the mediapipe file manually and then copy the path to your cmd terminal. Then you'll find that you installed mediapipe. Works on windows 64 bits

can you provide any screenshot of steps?

**Comment by Weq511762338** [comment]
If you are using Windows, according to the [Troubleshooting page](https://developers.google.com/mediapipe/framework/getting_started/troubleshooting) :
> after running pip install mediapipe usually indicates that there is no qualified MediaPipe Python for your system. Please note that MediaPipe Python PyPI officially supports the 64-bit version of Python 3.7 to 3.10 on the following OS:
x86_64 Linux
x86_64 macOS 10.15+
amd64 Windows
**If the OS is currently supported and you still see this error, please make sure that both the Python and pip binary are for Python 3.7 to 3.10**

I just installed 3.10 64-bit version and worked

**Comment by prankshtain** [comment]
I used python 3.9.7 and it worked 😍

**Comment by AXLShorts** [comment]
I had version Python 3.11.x and was getting this error. Installed version 3.7.0 and it just started working. Thanks everyone!

**Comment by Suckaaatit** [comment]
@AXLShorts U GOD BRO!!!

**Comment by NoureddinSameer** [comment]
I was working on a specific project that uses Mediapipe. However, as we know, the latest version of Python does not support Mediapipe. Therefore, if you have a project that uses Mediapipe, you should install Python version 3.9.13 or a lower version. I have other projects that run on the latest version of Python, which is 3.11.3. To solve this problem, I discovered that it's possible to have multiple versions of Python on the same operating system. I would like to express my gratitude to the person who helped me achieve this, Thank you Mr. [Konstantin Lübeck](https://de.linkedin.com/in/konstantin-luebeck?trk=public_post-text)
and I'm happy to share his video, which you can check out on YouTube:
[https://lnkd.in/dZTwikQN](https://lnkd.in/dZTwikQN?trk=public_post-text)

**Comment by feliipenevesnow** [comment]
I had to install version 3.7.0 for it to work 😅

**Comment by vaibhav1806** [comment]
> py -m pip install mediapie

still not working showing same thing ....

**Comment by AXLShorts** [comment]
This is a detailed troubleshoot I did on a number of devices after first success. ( Your mileage may vary)

Editor: VS Code (Versions Varied)
Install Command: pip stall mediapipe

At first I had python 3.11 installed and faced this error when installing mediapipe.

To be sure I had a clean environment, I fully uninstalled python from the system. I am not much of a python dev so I know there could have been another way. Anyways, after that I installed 3.7.0 ran the command everything worked but I had other libraries that weren't supported by 3.7.0 so I tested a bunch of version after that about 5 to 10 upto version 3.10.0.

Every single version worked without any error. After 3.10.0 I started facing same issue.

Then I tested something else, you can have multiple python versions installed and even when your project is already open, just change your interpreter to any version between 3.7.0 to 3.10.0 and it will work.

Note: Everytime I changed interpreter, I had to download all other libraries again too. Your mileage may vary.

I am sorry but I can't test all the versions. I have a repository named "ASL" that uses mediapipe. I used that for testing. It's public. It's not a final project. It's a work in progress.

**Comment by kbillore** [comment]
It works with python version 3.10.7 and not with 3.11.xx

**Comment by akashlomas** [comment]
Hi

I have Windows 11 Pro with Python 3.11.0 and pip 23.2.1.
When I run "pip install mediapipe-model-maker", I get the following error.

![2](https://github.com/google/mediapipe/assets/633607/5430e6ac-99a2-47c6-9294-ed3b2b0be38d)
![1](https://github.com/google/mediapipe/assets/633607/a14824d2-3a31-4662-b5e5-bcd59df04b21)

Kindly assist!

Thanks

**Comment by AXLShorts** [comment]
> Hi
> 
> I have Windows 11 Pro with Python 3.11.0 and pip 23.2.1. When I run "pip install mediapipe-model-maker", I get the following error.
> 
> ![2](https://user-images.githubusercontent.com/633607/256457346-5430e6ac-99a2-47c6-9294-ed3b2b0be38d.png) ![1](https://user-images.githubusercontent.com/633607/256457360-a14824d2-3a31-4662-b5e5-bcd59df04b21.png)
> 
> Kindly assist!
> 
> Thanks

Try with python 3.10.0

**Comment by flostr** [comment]
For me it was an older macOS version. `x86_64 macOS 10.15+` is required.

**Comment by gvdnikhil** [comment]
> I cant install mediapipe in Python There is no such a package

ERROR: Could not find a version that satisfies the requirement mediapipe (from versions: none)
ERROR: No matching distribution found for mediapipe


I solved the issue by installing pyton 3.10 version x64 installer from 
https://www.python.org/ftp/python/3.10.0/python-3.10.0-amd64.exe
and installed the mediapipe using pip command .


I have finally successfully installed the mediaipe.

![image](https://github.com/google/mediapipe/assets/86193313/11242fbc-d5bf-4f9e-875d-82cacd9a7649)

**Comment by amirih** [comment]
$ pip install mediapipe
Defaulting to user installation because normal site-packages is not writeable
ERROR: Could not find a version that satisfies the requirement mediapipe (from versions: none)
ERROR: No matching distribution found for mediapipe
$ python --version 
Python 3.12.0


Fedora Linux 39 (Workstation Edition)

**Comment by IslamSehil** [comment]
This does not work ,I have tried :

> `pip install mediapipe`

> `python -m pip install mediapipe`

> `pip install mediapipe-model-maker`

PS: I also try to use a Venv
windows 11
Python 3.12.1
pip 23.3.1

**Comment by amirih** [comment]
It seem we needed to use the python versions below 3.10. I am not sure it it still solves the problem, but for me changing the version worked.

**Comment by Divyansh723** [comment]
> This does not work ,I have tried :
> 
> > `pip install mediapipe`
> 
> > `python -m pip install mediapipe`
> 
> > `pip install mediapipe-model-maker`
> 
> PS: I also try to use a Venv windows 11 Python 3.12.1 pip 23.3.1

I am also facing the same problem what can I do

**Comment by mayank6940** [comment]
mediapipe only works on python 3.7 version.. use py 3.7 it will work surely

**Comment by Suchir2004** [comment]
Check have you installed dlib in system this also cause trouble while installing mediapipe

**Comment by shivakumardistproc** [comment]
I could install mediapipe (0.9.0.1).

I created fresh conda environment with Python version 3.7.16 and installed mediapipe (using pip install mediapipe).

With Python 3.12.1 it didn't work.

OS: Windows 11

I did not experiment with other Python or mediapipe versions as I am happy to work with current installation.

Hope it helps someone.

**Comment by mahsank111** [comment]
> If you get: ERROR: Could not find a version that satisfies the requirement mediapipe ERROR: No matching distribution found for mediapipe when trying "pip install mediapie"
> 
> Try "py -m pip install mediapie" insted, worked for me on windows!

*mediapipe"

**Comment by ayushgdev** [comment]
Hello @mahsank111 This issue thread is closed and not actively monitored by us. We would request you to open a new issue if you are facing problems with installation, with all necessary details  of OS, Python version, etc. as requested in MediaPipe issue template. 

@shivakumardistproc Just a side note: MediaPipe is not supported yet on Python v3.12. We support 3.8, 3.9, 3.10 and 3.11.

**Comment by mayank18mtnl** [comment]
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
ERROR: Could not find a version that satisfies the requirement mediapipe (from versions: none)
ERROR: No matching distribution found for mediapipe
Process returned with code 1

**Comment by mayank18mtnl** [comment]
python version : 3.9.2
rpi arch : 64
bullseye

**Comment by kavya-seeramsetty** [comment]
> > This does not work ,I have tried :
> > > `pip install mediapipe`
> > 
> > 
> > > `python -m pip install mediapipe`
> > 
> > 
> > > `pip install mediapipe-model-maker`
> > 
> > 
> > PS: I also try to use a Venv windows 11 Python 3.12.1 pip 23.3.1
> 
> I am also facing the same problem what can I do

Try changing the version of python 
I tried with python 3.8.10 
and then try with python -m pip install mediapipe

**Comment by DasunNethsara-04** [comment]
I can't install mediapipe
![image](https://github.com/google/mediapipe/assets/99202052/796e8b18-f122-4dc6-bd90-e58d47311881)

**Comment by kavya-seeramsetty** [comment]
First install python version 3.8 and then give it a try

**Comment by DasunNethsara-04** [comment]
> First install python version 3.8 and then give it a try

Ok, I'll try that. Thank you very much.

**Comment by paul-hernandez-herrera** [comment]
"pip install mediapipe" --- Works with Python 3.10 in Windows 11

**Comment by BlackieResources** [comment]
> I can't install mediapipe ![image](https://github.com/google/mediapipe/assets/99202052/796e8b18-f122-4dc6-bd90-e58d47311881)

This is what worked best for me because i was having your exact issue for awhile.


Go to this page https://pypi.org/project/mediapipe/#files
find the newest version for your platform and see which version of python it wants you to use. 
For example i had reset my pc last week and i downloaded 1.13.2 but the last version of mediapipe was for 3.12.9.

Hope this helps someone

**Comment by Suchir2004** [comment]
@BlackieResources first install dlib manually then try to install media pipe. Or you can use pycharm to Install library
